yt/visualization/fits_image.py

```diff
@@ -1381,21 +1397,55 @@ def __init__(
             image_res = (image_res, image_res)
         res = (image_res[0], image_res[1])
         if data_source is None:
-            source = ds
+            source = ds.all_data()
```

Comment:
seems like a good idea, but is there a reason for doing this change specifically in this PR that I am not seeing ?

Reply:
I can't remember why I did this anymore exactly--I somewhat recall an error of some sort

Reply:
ah, now I remember--I needed this to be a data object so I could call `_determine_fields`, which we weren't doing before.